packages/flutter/test/widgets/scrollable_selection_test.dart

```diff
@@ -0,0 +1,641 @@
+    expect(controller.offset, 0.0);
+    final RenderParagraph paragraph0 = tester.renderObject<RenderParagraph>(find.descendant(of: find.text('Item 0'), matching: find.byType(RichText)));
+    final RenderParagraph paragraph1 = tester.renderObject<RenderParagraph>(find.descendant(of: find.text('Item 1'), matching: find.byType(RichText)));
+    final RenderParagraph paragraph2 = tester.renderObject<RenderParagraph>(find.descendant(of: find.text('Item 2'), matching: find.byType(RichText)));
+    expect(paragraph0.selections[0], const TextSelection(baseOffset: 6, extentOffset: 0));
+    expect(paragraph1.selections[0], const TextSelection(baseOffset: 6, extentOffset: 0));
+    expect(paragraph2.selections[0], const TextSelection(baseOffset: 6, extentOffset: 0));
+
+    await gesture.up();
+  });
+
+  testWidgets('preserve selection when out of view.', (WidgetTester tester) async {
+    final ScrollController controller = ScrollController();
+    await tester.pumpWidget(MaterialApp(
+      home: SelectionArea(
+        selectionControls: materialTextSelectionControls,
+        child: ListView.builder(
+          controller: controller,
+          itemCount: 100,
+          itemBuilder: (BuildContext context, int index) {
+            return Text('Item $index');
+          },
+        ),
+      ),
+    ));
+
+    controller.jumpTo(2000);
+    await tester.pumpAndSettle();
+    expect(find.text('Item 50'), findsOneWidget);
+    RenderParagraph paragraph50 = tester.renderObject<RenderParagraph>(find.descendant(of: find.text('Item 50'), matching: find.byType(RichText)));
+    final TestGesture gesture = await tester.startGesture(textOffsetToPosition(paragraph50, 2), kind: ui.PointerDeviceKind.mouse);
+    addTearDown(gesture.removePointer);
+    await tester.pump();
+    await gesture.moveTo(textOffsetToPosition(paragraph50, 4));
+    await gesture.up();
+    expect(paragraph50.selections[0], const TextSelection(baseOffset: 2, extentOffset: 4));
+
+    controller.jumpTo(0);
+    await tester.pumpAndSettle();
+    expect(find.text('Item 50'), findsNothing);
+
+    controller.jumpTo(2000);
+    await tester.pumpAndSettle();
+    expect(find.text('Item 50'), findsOneWidget);
+    paragraph50 = tester.renderObject<RenderParagraph>(find.descendant(of: find.text('Item 50'), matching: find.byType(RichText)));
+    expect(paragraph50.selections[0], const TextSelection(baseOffset: 2, extentOffset: 4));
+
+    controller.jumpTo(4000);
+    await tester.pumpAndSettle();
+    expect(find.text('Item 50'), findsNothing);
+
+    controller.jumpTo(2000);
+    await tester.pumpAndSettle();
+    expect(find.text('Item 50'), findsOneWidget);
+    paragraph50 = tester.renderObject<RenderParagraph>(find.descendant(of: find.text('Item 50'), matching: find.byType(RichText)));
+    expect(paragraph50.selections[0], const TextSelection(baseOffset: 2, extentOffset: 4));
+  });
+
+  testWidgets('can select all non-mac', (WidgetTester tester) async {
```

Comment:
Nit: Maybe saying "Apple" instead of "Mac" is more accurate since it includes iOS.